atc/worker/client.go

```diff
@@ -1,16 +1,38 @@
 package worker
 
 import (
+	"code.cloudfoundry.org/garden"
 	"code.cloudfoundry.org/lager"
+	"context"
+	"fmt"
+	"github.com/concourse/concourse/atc"
 	"github.com/concourse/concourse/atc/db"
+	"io"
+	"path"
+	"strconv"
 )
 
+const taskProcessID = "task"
+const taskExitStatusPropertyName = "concourse:exit-status"
+
 //go:generate counterfeiter . Client
 
 type Client interface {
 	FindContainer(logger lager.Logger, teamID int, handle string) (Container, bool, error)
 	FindVolume(logger lager.Logger, teamID int, handle string) (Volume, bool, error)
 	CreateVolume(logger lager.Logger, vSpec VolumeSpec, wSpec WorkerSpec, volumeType db.VolumeType) (Volume, error)
+	RunTaskStep(
+		context.Context,
+		lager.Logger,
+		db.ContainerOwner,
+		ContainerSpec,
+		WorkerSpec,
```

Comment:
`WorkerSpec` and `ContainerPlacementStrategy` are Garden specific? are these fields optional?

Reply:
I don't see a way to avoid these for now. We may need to revisit this after we figure out how tags/teams and strategies are handled in k8s. We can make it optional for the k8s side of things.

`ContainerPlacementStrategy` might be able to be passed down at startup from `atccmd/command.go`. We could have a separate one for checks and for gets/puts/tasks, but this will depend on how all the other steps/checks are run.